api/firstbook.py

```diff
@@ -84,7 +85,7 @@ def remote_authenticate(self, username, password):
     # End implementation of BasicAuthenticationProvider abstract methods.
 
     def remote_pin_test(self, barcode, pin):
-        url = self.root + "&accesscode=%s&pin=%s" % tuple(map(
+        url = self.root + "/rest/V1/serialcode?code=%s&pin=%s" % tuple(map(
```

Comment:
Two comments here
1. How do you feel about using a newer string formatting approach here -- either `.format` or an f-string? Trade-offs of consistency v clarity I think
2. Can we store the path as a class variable? I think that's a pattern elsewhere and I like it for clarity's sake.